src/components/CatalogPage.tsx

```diff
@@ -0,0 +1,154 @@
+    if (selectedTypesArr.includes(resourceType)) {
+      setSelectedTypesArr(selectedTypesArr.filter((el) => el !== resourceType));
+    } else {
+      setSelectedTypesArr((prev) => {
+        return [...prev, resourceType];
+      });
+    }
+  };
+
+  console.log("SELECTED TYPES", selectedTypesArr);
+
+  const handleCheckedTagsBox = (tag: string) => {
+    if (selectedTagsArr.includes(tag)) {
+      setSelectedTagsArr(selectedTagsArr.filter((el) => el !== tag));
+    } else {
+      setSelectedTagsArr((prev) => {
+        return [...prev, tag];
+      });
+    }
+  };
+
+  return (
+    <div className="content">
+      <div className="ctn-catalog-page-left">
+        <input
+          className="catalog-search-input"
+          value={searchInput}
+          onChange={(e) => setSearchInput(e.target.value)}
+          placeholder="search here..."
+        ></input>
+
+        <div className="ctn-filter-by-type">
+          <h3 className="type-title"> FILTER BY TYPE...</h3>
+          {typesArr.map((type) => {
+            return (
+              <div key={type} className="nav-box">
+                <input
+                  type="checkbox"
+                  id="checkbox"
+                  value={type}
+                  onChange={() => handleCheckedTypeBox(type)}
+                />
+                <label htmlFor="checkbox">{type}</label>
+              </div>
+            );
+          })}
+        </div>
+        <div className="ctn-filter-by-tags">
+          <h3 className="tag-title"> FILTER BY TAG...</h3>
+          {tagsArr.map((tag) => {
+            return (
+              <div key={tag}>
+                <input
+                  type="checkbox"
+                  id="checkbox"
+                  value={tag}
+                  onChange={() => handleCheckedTagsBox(tag)}
+                />
+                <label htmlFor="checkbox"> {tag}</label>
+              </div>
+            );
+          })}
+        </div>
+      </div>
+
+      <div className="ctn-catalog-page-right">
+        {signedInUser && (
+          <Link to="/add_resource" className="add-resource-btn">
+            Add new resource
+          </Link>
+        )}
+
+        {signedInUser && <Link to="/add_resource"> </Link>}
+        {resources.length > 0 &&
+          filterResources(
+            searchInput,
+            selectedTypesArr,
+            selectedTagsArr,
+            resources
+          ).map((resource) => {
+            return (
+              <Resource
+                resourceData={resource}
+                signedInUser={signedInUser}
+                getSignedInUserLikes={getSignedInUserLikes}
+                fetchAndStoreResources={fetchAndStoreResources}
+                key={resource.id}
+                allUsers={allUsers}
+                studyListArr={studyListArr}
+                userLikes={userLikes}
+              />
+            );
+          })}
+      </div>
+    </div>
+  );
+}
+
```

Comment:
Overall clear file, maybe think about modularisation, perhaps some of the code in here could have been separate components?